### PR TITLE
Restart the DRB service when it completes it’s thread

### PR DESCRIPTION
**Pull request by JustinAiken**
Unless it’s explicitly stopped by the user

### CODE REVIEW

**Comment by JustinAiken** [comment]
Rebased with changelog and specs... should be ready to merge now, pending your review @benlangfeld

**Comment by JustinAiken** [comment]
Hmm, travis says no... I'll figure out why these are passing at home with different rubies and failing on Travis, better not merge this until CI is fixed.

**Comment by JustinAiken** [comment]
Yay green! Just needed a little pause to allow the DRB process time to restart; guess Travis's VMs aren't as speedy as a mighty Mac. 

Should be clear to merge now.

File: lib/adhearsion/drb/service.rb [hunk 1]
```diff
@@ -4,75 +4,98 @@
...
-      class << self
+      attr_accessor :user_stopped
 
-        ##
-        # Start the DRb service
-        def start
-          acl = create_acl config.acl.allow, config.acl.deny
+      def initialize
+        @user_stopped = false
+        Thread.abort_on_exception = true
```
Comment (benlangfeld) [inline lib/adhearsion/drb/service.rb]: I'm not so sure about including this here, since it's global. Necessary?

**Comment by benlangfeld** [comment]
Did you manage to test this in a real app (perhaps by implementing something in the server object which raises when called)?

**Comment by JustinAiken** [comment]
Raising exceptions in the server object never breaks the DRB thread that I can find.. but I tried setting up a method in the DRB object that closed the drb connection directly, and whenever calling that, this restarts it successfully in the real app.

File: lib/adhearsion/drb/service.rb (hunk 1 above)
Comment (JustinAiken) [inline lib/adhearsion/drb/service.rb]: Removed in commit below.

**Comment by benlangfeld** [comment]
Go ahead and merge/release as soon as Travis says yes. You have authorship on the gem.

**Pull request merged**